### DISCUSSION TITLE
2D physics on a plane?

### DISCUSSION THREAD

**Discussion by shenghan97** [pmndrs/use-cannon#375] [Q&A] [answered]
Hey!

I am experimenting with the idea of using r3f and cannon to make a 2D environment with 2D physics with an orthographic camera. Currently, I want to have a bunch of 2d circles collide with and bounce off each other. Any ideas of how to achieve this?

I do have an idea to use a cylinder to represent 2d circles. But the question is how to make them only move in collide in two detention (moving along the plane, in this case, the XY plane). _Is there any way in three.js or cannon to natively ignore the physics on the Z-axis?_  If there's not, I guess I could set up constraints on the z-axis or a huge amount of gravity on the z-axis. Would this be a good plan? 

Any feedback is welcome!! Thank you so much.

**Comment by joergjaeckel** [comment, marked as answer]
Try https://github.com/pmndrs/use-p2

**Reply by shenghan97** [reply]
This is fantastic! I don't know why I didn't find this earlier!

**Reply by shenghan97** [reply]
I just realized the keyword _2d physics_ and _react three fiber_ were not mentioned together in the readme of the repo at all, even though that's what it is about. Haha! Do you think it's a good idea to mention them closely together and help SEO and other people to find it?
